From 371172f4434c626856be83cc7ef27d6e912c3692 Mon Sep 17 00:00:00 2001
From: Miles Hu <milehu@codeaurora.org>
Date: Sat, 22 Jun 2019 12:58:35 -0700
Subject: [PATCH 1/2] ath11k: fix no traffic problem if co-exist monitor is
 enabled.

Clear monitor flag for regular data packet CB to avoid drop in mac80211

Signed-off-by: Miles Hu <milehu@codeaurora.org>
---
 drivers/net/wireless/ath/ath11k/dp_rx.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/net/wireless/ath/ath11k/dp_rx.c
+++ b/drivers/net/wireless/ath/ath11k/dp_rx.c
@@ -4002,6 +4002,7 @@ static int ath11k_dp_rx_mon_deliver(stru
 
 	header = mon_skb;
 
+	rxs->flag = 0;
 	do {
 		skb_next = mon_skb->next;
 		if (!skb_next)
@@ -4023,6 +4024,7 @@ static int ath11k_dp_rx_mon_deliver(stru
 		ath11k_dp_rx_deliver_msdu(ar, napi, mon_skb);
 		mon_skb = skb_next;
 	} while (mon_skb && (mon_skb != tail_msdu));
+	rxs->flag = 0;
 
 	return 0;
 
